game/scripts/vscripts/Sections/Chapter5/SectionTeamFight.ts

```diff
@@ -0,0 +1,183 @@
+            tg.loop(ctx => shared.getLivingEnemyHeroes(ctx).length > 0, ctx => tg.seq([
+                tg.immediate(_ => goalKillEnemyHeroes.setValue(5 - shared.getLivingEnemyHeroes(ctx).length)),
+                tg.wait(1),
+            ])),
+            tg.immediate(_ => goalKillEnemyHeroes.setValue(5)),
+            tg.immediate(_ => goalKillEnemyHeroes.complete()),
+
+            // Play win dialog
+            tg.immediate(ctx => shared.getLivingFriendlyHeroes(ctx).forEach(hero => hero.StartGesture(GameActivity.DOTA_VICTORY))),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_5, LocalizationKey.Script_5_5v5_5, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+
+            // Add tp scroll
+            tg.immediate(_ => playerHero.AddItemByName("item_tpscroll").EndCooldown()),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_6, LocalizationKey.Script_5_5v5_6, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_7, LocalizationKey.Script_5_5v5_7, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_8, LocalizationKey.Script_5_5v5_8, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+
+            // Wait for player to try to use the tp
+            tg.immediate(_ => waitingForPlayerTp = true),
+            tg.completeOnCheck(_ => playerUsedTp, 0.1),
+
+            // More dialog about importance of tps and bait player into using voice
+            tg.audioDialog(LocalizationKey.Script_5_5v5_9, LocalizationKey.Script_5_5v5_9, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_10, LocalizationKey.Script_5_5v5_10, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+            tg.audioDialog(LocalizationKey.Script_5_5v5_11, LocalizationKey.Script_5_5v5_11, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+
+            // Wait for player to use their voice, or not
+            tg.immediate(_ => goalPromiseCarryTp.start()),
+            tg.forkAny([
+                tg.seq([
+                    tg.waitForCommand(37), // Team voice
+                    tg.immediate(_ => voicePressed = true),
+                    tg.audioDialog(LocalizationKey.Script_5_5v5_12, LocalizationKey.Script_5_5v5_12, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
+                ]),
+                tg.seq([
+                    tg.wait(6),
+                    tg.completeOnCheck(_ => !voicePressed, 0.1),
+                    tg.audioDialog(LocalizationKey.Script_5_5v5_13, LocalizationKey.Script_5_5v5_13, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
+                ])
+            ]),
+            tg.immediate(_ => goalPromiseCarryTp.complete()),
+
+            // Actually tp the player to the fountain
+            tg.immediate(_ => freezePlayerHero(true)),
+            tg.immediate(_ => playerHero.RemoveItem(getOrError(playerHero.FindItemInInventory("item_tpscroll"), "Could not find tp scroll"))),
```

Comment:
I'm not entirely sold on the fake TP thing, let's make them actually TP.
If needed, maybe rearrange the script so you first promise to always have a TP, then TP? Not jebaits. Might be better.

Reply:
Promise to do that later, just wanted to get something functioning up! Also jebait is too fun to remove.